source/include/core_json.h

```diff
@@ -155,15 +156,17 @@ JSONStatus_t JSON_Validate( const char * buf,
  *         value[ valueLength ] = save;
  *     }
  * @endcode
+ *
+ * @note The maximum index value is ~2 billion ( 2^31 - 9 ).
  */
 /* @[declare_json_search] */
 JSONStatus_t JSON_Search( char * buf,
                           size_t max,
                           const char * queryKey,
                           size_t queryKeyLength,
-                          char separator,
                           char ** outValue,
                           size_t * outValueLength );
 /* @[declare_json_search] */
+#define MAX_INDEX_VALUE    ( 0x7FFFFFF7 )   /* 2^31 - 9 */
```

Comment:
The max index value has to be passed as an ASCII string to the function. Does providing the max index value as an integer value provide use for the customer? (Should it be instead provided as an ASCII string?)

Reply:
Really, no one should be iterating over 2 billion elements into a JSON array.  I originally put in the calculation seen in the comment, but struggle with the MISRA complaints that ensued.

Reply:
What is the utility of exposing this macro in the API if the user _should_ not be using it. Can we just move it to within the source file?

Reply:
I reference the macro in tests.